tutorials/time-freq/ssvep.py

```diff
@@ -182,212 +194,516 @@ def snr_spectrum(psd, noise_n_neighborfreqs=1, noise_skip_neighborfreqs=1):
 
-# get grand average SNR per channel (all subs, all trials) and electrode labels
-snr_grave = snrs_stim.mean(axis=0)
+# add xyz coordinates for all channels
+montage.positions = montage._get_ch_pos()  # luckily i dug this out in the mne code!
 
-# select only present channels from the standard montage
+# select only channels from the standard montage that are present in our data
 topo_pos_grave = []
 [topo_pos_grave.append(montage.positions[ch][:2]) for ch in epochs.info['ch_names']]
 topo_pos_grave = np.array(topo_pos_grave)
 
-# plot SNR topography
+# plot SNR topography, eventually
 f, ax = plt.subplots()
-mne.viz.plot_topomap(snr_grave, topo_pos_grave, vmin=1., axes=ax)
-print("sub 2, all trials")
-print("average SNR: %f" % snr_grave.mean())
+mne.viz.plot_topomap(snrs_12hz_chaverage, topo_pos_grave, vmin=1., axes=ax)
 
-###############################################################################
-# Subsetting data
-# ---------------
+print("sub 2, 12hz trials, SNR at 12hz")
+print("average SNR (all channels): %f" % snrs_12hz_chaverage.mean())
+print("average SNR (occipital ROI): %f" % snrs_target.mean())
+
+tstat_roi_vs_scalp = ttest_rel(snrs_target.mean(axis=1), snrs_12hz.mean(axis=1))
+print("12 hz SNR in occipital ROI is significantly larger than 12 hz SNR over all channels"
+      ": t = %.3f, p = %f" % tstat_roi_vs_scalp)
+
+
+##############################################################################
+# We can see, that 1) this participant indeed exhibits a cluster of chanels
+# with high SNR in the occipital region and 2) that the average SNR over all
+# channels is smaller than the average of the visual ROI computed above.
+# The difference is statistically significant. Great!
+#
+# Such a topoplot can be a nice tool to explore and play with your data - e.g.
+# you could try how changing the reference will affect the spatial
+# distribution of SNR values.
 #
-# For statistical comparison you probably want specific subsets of the SNR
-# array. Either some channels, or - obviously - different trials depending on
-# the stimuli.
+# However, we also wanted to show this plot to illustrate
+# a large problem with frequency-tagged or any other brain imaging data:
+# there are many channels and somewhere you will likely find some significant
+# effects. It's very easy - even unintended - to end up
+# double-dipping or p-hacking.
+# Avoid this either by selecting your ROI or channels for analysis *a priori*
+# (and ideally preregister this decision, so people will believe you), or if
+# you select an ROI or individual channel for reporting *because this channel
+# or ROI shows an effect* do so transparently and correct for multiple
+# comparison.
+
+
+##############################################################################
+# Statistical analysis
+# --------------------
+# Now that we finished this edgy little open science lesson let's move on and
```

Comment:
I am not sure what you mean by that. I get that it is a joke but not sure what is about. Are using "edgy" sarcastically because this is supposed to be common knowledge?